src/Microsoft.AspNetCore.Authorization/DefaultAuthorizationPolicyProvider.cs

```diff
@@ -45,7 +45,8 @@ public Task<AuthorizationPolicy> GetDefaultPolicyAsync()
         /// <returns>The named <see cref="AuthorizationPolicy"/>.</returns>
         public virtual Task<AuthorizationPolicy> GetPolicyAsync(string policyName)
         {
-            // MVC relies on DefaultAuthorizationPolicyProvider providing the same policy for the same requests.
+            // MVC relies on this class providing the same policy for the same requests. If it were to provide
+            // different policies per request then the cached policy in MVC could cause undesired access.
```

Comment:
Nit: Maybe something more like?

MVC caches policies specifically for this class, so this method must return the same policy for every request or it could allow undesired access.